lib/internal/inspector_async_hook.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const { createHook } = require('async_hooks');
+const inspector = process.binding('inspector');
+const errors = require('internal/errors');
+
+if (!inspector || !inspector.asyncTaskScheduled) return;
+
+module.exports = createHook({
+  init(asyncId, type, triggerId, resource) {
+    if (!Number.isInteger(asyncId))
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'asyncId', 'integer');
+    if (typeof type !== 'string')
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'type', 'string');
+    // TODO(bajtos): support recurring tasks like setInterval
```

Comment:
What is the consequence if we just set it to `true` for all event types?

Using `false` does not make sense because of `setInterval` etc.. But even for `setInterval` it can't be guaranteed that it will be `recurring`. Thus `true` should always work, although it might be a bit inefficient.